### PR TITLE
Edit ride works

### PR DESCRIPTION
**Pull request by DarNachmani**

### CODE REVIEW

File: apps/ui/src/apis/rides/edit-ride.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  try {
+    const responseData = await graphqlRequest<GraphQLRideResponse>(query);
+    return responseData.updateRideById.ride;
+  } catch (error) {
+    console.error('Error updating ride:', error);
+    throw error;
+  }
+};
+
+const postUpdateRiders = async (
```
Comment (Kovler10) [inline apps/ui/src/apis/rides/edit-ride.ts:34]: maybe `cancelRideByRider` would be a better name here

File: apps/ui/src/apis/rides/edit-ride.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import { Ride } from '@communetypes/Ride';
+import { graphqlRequest } from '../graphql';
+import { EditRideSchema } from '@communetypes/EditRideSchema';
+import { updateRideQuery, updateRidersQuery } from '../utils/userRideQueries';
+import { Rider } from '@communetypes/Rider';
```
Comment (Kovler10) [inline apps/ui/src/apis/rides/edit-ride.ts:5]: why 3 imports from local package?

File: apps/ui/src/Components/Rides/RideCard/index.tsx [hunk 3]
```diff
@@ -10,21 +10,31 @@ import {
...
 import { RideDetails } from '../RideDetails';
 import { JoinRideDialog } from '../JoinRide';
 import { rideStatusIcons } from '../../../utils/communities/userStatusIcons';
+import { useUser } from '../../../hooks/Users/useUser';
+import { EditRideDialog } from '../EditRide';
 
 interface RideCardProps {
   ride: Ride;
   rideStatus: UserRide | undefined;
+  communities: Community[];
```
Comment (Kovler10) [inline apps/ui/src/Components/Rides/RideCard/index.tsx:26]: maybe a context would be better

File: apps/ui/src/Components/Rides/EditRide/index.tsx [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+      setDestination(location);
+    }
+  };
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const handleSubmit = async () => {
+    if (!community || !startLocation || !destination || !gasMoney || !seats) {
+      alert('All fields are required.');
```
Comment (Kovler10) [inline apps/ui/src/Components/Rides/EditRide/index.tsx:84]: use a snackbar instead

File: apps/ui/src/Components/Rides/EditRide/index.tsx [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+      png: ride.png,
+    };
+
+    await editRide(updatedRide);
+    if (isSuccess) {
+      handleClose();
+    }
+  };
+
+  if (isSuccess) {
```
Comment (Kovler10) [inline apps/ui/src/Components/Rides/EditRide/index.tsx:118]: either this if or the one inside the function is redundant

**Review by Kovler10 — CHANGES_REQUESTED** [review]

File: apps/ui/src/apis/rides/edit-ride.ts (hunk 2 above)
Comment (DarNachmani) [reply apps/ui/src/apis/rides/edit-ride.ts:5]: What

File: apps/ui/src/Components/Rides/RideCard/index.tsx (hunk 3 above)
Comment (DarNachmani) [reply apps/ui/src/Components/Rides/RideCard/index.tsx:26]: Why?

File: apps/ui/src/Components/Rides/EditRide/index.tsx [hunk 6]
```diff
@@ -110,9 +114,9 @@ const EditRideDialog = ({
     };
 
     await editRide(updatedRide);
-    if (isSuccess) {
-      handleClose();
-    }
+    // if (isSuccess) {
```
Comment (Kovler10) [inline apps/ui/src/Components/Rides/EditRide/index.tsx:117]: please remove this comment

File: apps/ui/src/apis/rides/edit-ride.ts (hunk 2 above)
Comment (Kovler10) [reply apps/ui/src/apis/rides/edit-ride.ts:5]: you have 3 different import from the types package

File: apps/ui/src/Components/Rides/EditRide/index.tsx (hunk 4 above)
Comment (Kovler10) [reply apps/ui/src/Components/Rides/EditRide/index.tsx]: this did not change

**Review by Kovler10 — CHANGES_REQUESTED** [review]

File: apps/ui/src/Components/Rides/EditRide/index.tsx (hunk 4 above)
Comment (DarNachmani) [reply apps/ui/src/Components/Rides/EditRide/index.tsx]: Yes it did

File: apps/ui/src/apis/rides/edit-ride.ts (hunk 2 above)
Comment (DarNachmani) [reply apps/ui/src/apis/rides/edit-ride.ts:5]: So what? I need them

**Review by Kovler10 — APPROVED** [review]

**Pull request merged**